r4/metadata/tests/test_service.bal

```diff
@@ -0,0 +1,92 @@
+
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/test;
+import health.fhir.templates.r4.metadata.constants;
+import wso2healthcare/healthcare.fhir.r4;
+
+# A resource for generating capability statement
+# + return - capability statement as a json
+function testError() returns r4:FHIRError { 
+    error response = error("Test Error");
+    return r4:createFHIRError(response.message(), r4:FATAL, r4:TRANSIENT_EXCEPTION, response.detail().toString(), cause = response);
+}
+
+@test:Config { groups: ["Service"] }
+function testService() returns error? {
+    http:Client clientEndpoint = check new ("http://localhost:9090");
+    json|error response = clientEndpoint->get("/fhir/r4/metadata");
+    if response is json {
+        test:assertTrue(true);
+    } else {
+        test:assertTrue(false);
+    }
+}
+
+@test:Config { groups: ["Service"] }
+function testServiceFatalError() returns error? {
+    http:Client clientEndpoint = check new ("http://localhost:9090");
+    json|error response = clientEndpoint->get("/fhir/r4/metadata");
+    if response is json {
+        test:assertTrue(true);
+    } else {
+        test:assertTrue(false);
+    }
+}
+
+@test:Config { groups: ["Service"] }
+function testServiceWithSample() returns error? {
+    r4:CapabilityStatement sampleCapabilityStatement = {
+        resourceType: constants:CAPABILITY_STATEMENT,
+        status: "active",
+        kind: "instance",
+        fhirVersion: "4.0.1",
+        date: "23/03/2022",
+        format: ["json"],
+        patchFormat: [],
+        rest: []
+    };
+    http:Client clientEndpoint = test:mock(http:Client);
+    test:prepare(clientEndpoint).when("get").thenReturn(sampleCapabilityStatement);
+    json|error response = clientEndpoint->get("/fhir/r4/metadata");
+    test:assertEquals(response, sampleCapabilityStatement.toJson());
+}
+
+public function mockGenerateCapabilityStatement() returns json|error {
+    r4:CapabilityStatement|error capabilityStatement = mockReadFromConfigs();
+    if capabilityStatement is error {
+        return capabilityStatement;
+    } else {
+        return capabilityStatement.toJson();
+    }
+}
+
+# method to build capability statement from metadata configurables
+# + return - capabilitity statement json object
+function mockReadFromConfigs() returns r4:CapabilityStatement|error {
+    r4:CapabilityStatement capabilityStatement = {
+        resourceType: constants:CAPABILITY_STATEMENT,
+        status: "active",
+        kind: "instance",
+        fhirVersion: "4.0.1",
+        date: "23/03/2022",
+        format: [],
+        patchFormat: [],
+        rest: []
+    };
+    return capabilityStatement;
```

Comment:
No need to create a variable.

```suggestion
    return {
        resourceType: constants:CAPABILITY_STATEMENT,
        status: "active",
        kind: "instance",
        fhirVersion: "4.0.1",
        date: "23/03/2022",
        format: [],
        patchFormat: [],
        rest: []
    };
```

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerinax-health.fhir.templates/commit/8dacef8d0685638ff61ce7cb1398abe9236ed812